catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/MigrationFileWriter.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.migratable.impl;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.util.List;
+
+import org.codice.ddf.migration.MigrationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.MetacardImpl;
+
+/**
+ * Responsible for all low-level IO operations pertaining to migration.
+ */
+public class MigrationFileWriter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MigrationFileWriter.class);
+
+    //      ========================================================
+    //      TODO: Look into moving these functions to MigratableUtil
+    //      ========================================================
+
+    /**
+     * Creates the directory for the catalog export if it doesn't exist.
+     * <p>
+     * If the directory does exist, work based on the assumption it was cleaned by the parent
+     * process in the management logic of inport and export. There is no point to delegate
+     * low level directory operations to each and every migratable when they can be abstracted
+     * into the base logic.
+     *
+     * @param exportPath The path representing the directory to create.
+     * @throws MigrationException thrown if the export directory doesn't exist and could not be created
+     */
+    public void init(Path exportPath) throws MigrationException {
+        final File exportDir = exportPath.toFile();
+        if (!exportDir.exists() || !exportDir.isDirectory()) {
```

Comment:
We should consider using `FileUtils.forceMkdir()` instead of doing all those ugly `File` checks. Otherwise we'll also need to make sure that directory can be written to.